static/app/components/forms/selectControl.tsx

```diff
@@ -344,8 +345,12 @@ function SelectControl<OptionType extends GeneralSelectValue = GeneralSelectValu
       value={mappedValue}
       isMulti={props.multiple || props.multi}
       isDisabled={props.isDisabled || props.disabled}
+      showDividers={props.showDividers}
       options={options || (choicesOrOptions as OptionsType<OptionType>)}
       openMenuOnFocus={props.openMenuOnFocus === undefined ? true : props.openMenuOnFocus}
+      blurInputOnSelect={!(props.multiple || props.multi)}
```

Comment:
I would just write this as 
```suggestion
      blurInputOnSelect={!props.multiple && !props.multi}
```